v1/nix/pkgs/fetchPipMetadata/src/fetch_pip_metadata/lock_file_from_report.py

```diff
@@ -211,7 +213,10 @@ def lock_file_from_report(report):
             evaluate_requirements(
                 env, requirements, dependencies, root_name, extras, list()
             )
-            targets[extra] = dependencies
+            if extra not in targets:
+                targets[extra] = dependencies
+            else:
+                targets[extra].update(dependencies)
```

Comment:
Ah, as easy as that! Thanks for looking into it, seems to be the right fix here.